crates/zksync_os_fri_prover/src/lib.rs

```diff
@@ -150,6 +150,11 @@ pub async fn run(args: Args) -> anyhow::Result<()> {
 
     let mut proof_count = 0;
 
+    let mut retry_count = 0;
+    let retry_interval = Duration::from_millis(100);
```

Comment:
Wouldn't it be easier to log every Xth execution, instead of doing it time based? And if we really want to do it time based, why not start a time counter? If `time.elapsed() > retry_log_interval`, reset counter & log. This way we avoid random arithmetics with numbers & contamination (if it did run for 10 times and retry duration was 100ms, but turns out it did some execution elsewhere and instead of 5m, it's more like 7m?)

Also, I'd like us to still have some logs (maybe they can be in trace/debug mode).

Reply:
which execution? When an execution is actually done we reset the counter. I don't really want to do the plain counters, since we may change the waiting interval, but I like the suggestion about time counter.

Reply:
Execution of code, AKA `run_inner`.